git/internal/e2e/run.sh

```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+# This script runs e2e tests for pkg/git.
+
+set -o errexit
+PROJECT_DIR=$(git rev-parse --show-toplevel)
+DIR="$(cd "$(dirname "$0")" && pwd)"
+
+cd "${PROJECT_DIR}/git/libgit2" && make libgit2
+
+source "${DIR}/setup_gitlab.sh"
+
+LIBGIT2_IMG=ghcr.io/fluxcd/golang-with-libgit2
+LIBGIT2_TAG=v0.2.0
+LIBGIT2_PKG_CONFIG_PATH="${PROJECT_DIR}/git/libgit2/build/libgit2/${LIBGIT2_TAG}/lib/pkgconfig"
+CGO_LDFLAGS=$(PKG_CONFIG_PATH="${LIBGIT2_PKG_CONFIG_PATH}" pkg-config --libs --static --cflags libgit2)
```

Comment:
These variables are now being created here and in Makefile of libgit2 package.
Would it be better to put these in some script and source and populate the values in env vars?
That way, we have one source of these values, less scattered, easy to maintain in the long term.

Reply:
Maybe this applies to the other variables like the image and tag. Updating in one file, would update both libgit2 package and this package.

Reply:
I think that the the value of consolidating the variables to improve long term maintenance outweighs the other concerns.

Regarding "bash getting cumbersome", if it's a simple script with the expectation to be sourced. I think it would not contain much more than e.g. `export`, `declare` and _maybe_ `readonly` declarations. Which should be fairly easy to understand (the scope of), and extend.